gpt_index/readers/slack.py

```diff
@@ -36,14 +37,36 @@ def __init__(self, slack_token: Optional[str] = None) -> None:
         if not res["ok"]:
             raise ValueError(f"Error initializing Slack API: {res['error']}")
 
+    def _read_message(self, channel_id: str, message_ts: str) -> str:
+        """Read a message."""
+
+        messages_text = []
+        next_cursor = None
+        while True:
+            # https://slack.com/api/conversations.replies
+            # List all replies to a message, including the message itself.
+            result = self.client.conversations_replies(
+                channel=channel_id, ts=message_ts, cursor=next_cursor)
+            try:
+                messages = result["messages"]
+                for message in messages:
+                    messages_text.append(message["text"])
+
+                if not result["has_more"]:
+                    break
+
+                next_cursor = result["response_metadata"]["next_cursor"]
+            except SlackApiError as e:
+                logger.error("Error parsing conversation replies: {}".format(e))
+
+        return "\n\n".join(messages_text)
+
     def _read_channel(self, channel_id: str) -> str:
         """Read a channel."""
-        from slack_sdk.errors import SlackApiError
```

Comment:
ah i'd probably leave the import here (this allows people who haven't installed slack_sdk to still use GPT index)

Reply:
That makes sense. Reverted it.